python/lsst/daf/butler/core/utils.py

```diff
@@ -307,3 +313,38 @@ def __setstate__(self: Any, state: Any) -> None:  # noqa: N807
         assert not state
     cls.__setstate__ = __setstate__
     return cls
+
+
+def findFiles(values: Iterable[str], regex: Optional[str] = None) -> List[str]:
```

Comment:
This is going to have to be changed soon to use ButlerURI -- not for this ticket -- but maybe for now the name should be `findFileResources` or something to indicate that we aren't always going to be local files.

Reply:
Yes, you said in the middleware meeting that this impl would not work for e.g. S3 but that it's good enough for now. The name change is a good idea, will do.